content/blog/eng/careful-with-that-constant-developer.md

```diff
@@ -0,0 +1,462 @@
+}
+
+namespace {
+    echo Jedi\SIDE_OF_THE_FORCE;
+    // light
+    echo Sith\SIDE_OF_THE_FORCE;
+    // dark
+}
+```
+
+Notice the unusual namespace bracketed syntax, it's the only way for combining into a single file global non-namespaced code with namespaced code. Anyway, this example, it's just for demonstration purposes, because it's strongly discouraged as a coding practice to combine multiple namespaces into the same file.  
+Notice also that since you are out of any class hierarchy, you are not allowed to use access modifiers for `const` that is therefore set to a public default visibility. 
+
+# Sensing the future
+
+All the scenarios we've briefly seen until now could be addressed differently, thinking of one of the new PHP features.  
+Although other languages featured Enumerations for a long time, they are going to be available in PHP only since version 8.1.  
+And it's a fact that [Enumerations](https://wiki.php.net/rfc/enumerations) (enumerated types with a fixed number of possible values) offer other interesting implementation possibilities. Maybe some way to reconsider the use of constants too.  
+Take a look at the following code  
+
+```php
+<?php
+declare(strict_types=1);
+
+enum Force: string 
+{
+    case LIGHT_SIDE = 'light';
+    case DARK_SIDE = 'dark';
+}
+
+echo Force::LIGHT_SIDE->value;
+// light
+```
+
+Why is that interesting? For example, because Enumerations cases are represented as constants on the enum itself and their values are read-only properties.  
+It is worth to say that Enumerations are much more than that, since they are built on top of classes and objects, so they can have their own constants and methods too, and also they can implement interfaces.  
+So maybe, coming back to what we were talking about when we were dealing with class hierarchies, something like the following code does the trick
+
+```php
+<?php
+declare(strict_types=1);
+
+enum Force
+{
+    case LIGHT_SIDE;
+    case DARK_SIDE;
+    
+    public function control(): string 
+    {
+        return match($this) {
+            self::LIGHT_SIDE => 'light',   
+            self::DARK_SIDE => 'dark'
+        }
+    }
+}
+
+class Jedi
+{
+    private Force $force;
+
+    public __construct(Force $force) 
+    {
+        $this->force = $force;
+    }
+
+    public function useTheForce(): string 
+    {
+        return $this->force->control();
+    }
+}
+
+$obiwan = new Jedi(Force::LIGHT_SIDE);
+echo $obiwan->useTheForce();
+// light
+$anakin = new Jedi(Force::DARK_SIDE);
+echo $anakin->useTheForce();
+// dark
+```
+
+# Conclusion
+
+Constants should be the last concern during development, in PHP too. Nonetheless, they carry with them a lot of situations that a developer needs to consider.  
+There are a few things to pay attention to, but for sure, the amount of possibilities that PHP gives, puts the developer in a real creative position, with the concrete possibility to make consistent use of constants.  
```

Comment:
There are a few things to pay attention to but, for sure, the amount of possibilities that PHP gives puts ..